packages/flutter/lib/src/cupertino/tab_scaffold.dart

```diff
@@ -540,15 +540,18 @@ class _TabSwitchingViewState extends State<_TabSwitchingView> {
         final bool active = index == widget.currentTabIndex;
         shouldBuildTab[index] = active || shouldBuildTab[index];
 
-        return Offstage(
-          offstage: !active,
-          child: TickerMode(
-            enabled: active,
-            child: FocusScope(
-              node: tabFocusNodes[index],
-              child: Builder(builder: (BuildContext context) {
-                return shouldBuildTab[index] ? widget.tabBuilder(context, index) : Container();
-              }),
+        return HeroMode(
```

Comment:
This seems like a bandage fix.

I am not sure if this make sense, but should we apply HeroMode in Offstage directly? For example if Offstage.offstage = true, the it will wrap its child with HeroMode. 

cc @goderbauer @justinmc

Reply:
No, it should be composable. Just like Offstage also doesn't wrap TickerMode and some other widgets.

Reply:
sounds good